### PR TITLE
[Enhancement] add batch insert statistics on fe

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
1. full statistics collect will save statistics data in fe’s buffer
2. fe will generate `insert` sql written the data to BE when the buffer is full

* add hll serialized/deserialized function
* update full statistics collect logical
* update statistics config name

## Test case:
1. I create 20s table on flat ssb table with large partition by month, like that:
```
CREATE TABLE `lineorder_flat_x1` (
  `lo_orderdate` date NOT NULL COMMENT "",
  `lo_orderkey` int(11) NOT NULL COMMENT "",
  `lo_linenumber` tinyint(4) NOT NULL COMMENT "",
  `lo_custkey` int(11) NOT NULL COMMENT "",
  `lo_partkey` int(11) NOT NULL COMMENT "",
  `lo_suppkey` int(11) NOT NULL COMMENT "",
  `lo_orderpriority` varchar(100) NOT NULL COMMENT "",
  `lo_shippriority` tinyint(4) NOT NULL COMMENT "",
  `lo_quantity` tinyint(4) NOT NULL COMMENT "",
  `lo_extendedprice` int(11) NOT NULL COMMENT "",
  `lo_ordtotalprice` int(11) NOT NULL COMMENT "",
  `lo_discount` tinyint(4) NOT NULL COMMENT "",
  `lo_revenue` int(11) NOT NULL COMMENT "",
  `lo_supplycost` int(11) NOT NULL COMMENT "",
  `lo_tax` tinyint(4) NOT NULL COMMENT "",
  `lo_commitdate` date NOT NULL COMMENT "",
  `lo_shipmode` varchar(100) NOT NULL COMMENT "",
  `c_name` varchar(100) NOT NULL COMMENT "",
  `c_address` varchar(100) NOT NULL COMMENT "",
  `c_city` varchar(100) NOT NULL COMMENT "",
  `c_nation` varchar(100) NOT NULL COMMENT "",
  `c_region` varchar(100) NOT NULL COMMENT "",
  `c_phone` varchar(100) NOT NULL COMMENT "",
  `c_mktsegment` varchar(100) NOT NULL COMMENT "",
  `s_name` varchar(100) NOT NULL COMMENT "",
  `s_address` varchar(100) NOT NULL COMMENT "",
  `s_city` varchar(100) NOT NULL COMMENT "",
  `s_nation` varchar(100) NOT NULL COMMENT "",
  `s_region` varchar(100) NOT NULL COMMENT "",
  `s_phone` varchar(100) NOT NULL COMMENT "",
  `p_name` varchar(100) NOT NULL COMMENT "",
  `p_mfgr` varchar(100) NOT NULL COMMENT "",
  `p_category` varchar(100) NOT NULL COMMENT "",
  `p_brand` varchar(100) NOT NULL COMMENT "",
  `p_color` varchar(100) NOT NULL COMMENT "",
  `p_type` varchar(100) NOT NULL COMMENT "",
  `p_size` tinyint(4) NOT NULL COMMENT "",
  `p_container` varchar(100) NOT NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`lo_orderdate`, `lo_orderkey`)
COMMENT "olap"
PARTITION BY RANGE(`lo_orderdate`)
(
PARTITION p0 VALUES [("0000-01-01"), ("1993-01-01")),
PARTITION p1 VALUES [("1993-01-01"), ("1993-02-01")),
PARTITION p2 VALUES [("1993-02-01"), ("1993-03-01")),
PARTITION p3 VALUES [("1993-03-01"), ("1993-04-01")),
..........
PARTITION p82 VALUES [("1999-10-01"), ("1999-11-01")),
PARTITION p83 VALUES [("1999-11-01"), ("1999-12-01"))
)
DISTRIBUTED BY HASH(`lo_orderkey`) BUCKETS 48
PROPERTIES (
"replication_num" = "1",
"colocate_with" = "groupxx1",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false",
"replicated_storage" = "true",
"compression" = "LZ4"
);
```

2. load 10 rows data on every partitions, the statistics table should be 69 partition * 38 columns * 20 tables = 52440 rows least 
3. execute `anlyze full table` on all tables.

before this PR: version (1 -> 524265), row(54264)
```
MySQL hk_test> show tablet from _statistics_.column_statistics;
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
| TabletId | ReplicaId | BackendId | SchemaHash | Version | VersionHash | LstSuccessVersion | LstSuccessVersionHash | LstFailedVersion | LstFailedVersionHash | LstFailedTime | DataSize | RowCount | State  | LstConsistencyCheckTime | CheckVersion | CheckVersionHash | VersionCount | PathHash | MetaUrl                                           | CompactionStatus                                                |
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
| 191154   | 191155    | 10075     | 0          | 1       | 0           | 1                 | 0                     | -1               | 0                    | <null>        | 0B       | 0        | NORMAL | <null>                  | -1           | 0                | 0            | 0        | http://172.26.197.102:8040/api/meta/header/191154 | http://172.26.197.102:8040/api/compaction/show?tablet_id=191154 |
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
MySQL hk_test> show tablet from _statistics_.column_statistics;
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
| TabletId | ReplicaId | BackendId | SchemaHash | Version | VersionHash | LstSuccessVersion | LstSuccessVersionHash | LstFailedVersion | LstFailedVersionHash | LstFailedTime | DataSize | RowCount | State  | LstConsistencyCheckTime | CheckVersion | CheckVersionHash | VersionCount | PathHash | MetaUrl                                           | CompactionStatus                                                |
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
| 191154   | 191155    | 10075     | 0          | 54265   | 0           | 54265             | 0                     | -1               | 0                    | <null>        | 2.5MB    | 54264    | NORMAL | <null>                  | -1           | 0                | 1            | 0        | http://172.26.197.102:8040/api/meta/header/191154 | http://172.26.197.102:8040/api/compaction/show?tablet_id=191154 |
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
```

after this PR: version (1 -> 70), row(54264)
```
MySQL hk_test> show tablet from _statistics_.column_statistics;
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
| TabletId | ReplicaId | BackendId | SchemaHash | Version | VersionHash | LstSuccessVersion | LstSuccessVersionHash | LstFailedVersion | LstFailedVersionHash | LstFailedTime | DataSize | RowCount | State  | LstConsistencyCheckTime | CheckVersion | CheckVersionHash | VersionCount | PathHash | MetaUrl                                           | CompactionStatus                                                |
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
| 245484   | 245485    | 10076     | 0          | 1       | 0           | 1                 | 0                     | -1               | 0                    | <null>        | 0B       | 0        | NORMAL | <null>                  | -1           | 0                | 0            | 0        | http://172.26.197.103:8040/api/meta/header/245484 | http://172.26.197.103:8040/api/compaction/show?tablet_id=245484 |
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
MySQL hk_test> show tablet from _statistics_.column_statistics;
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
| TabletId | ReplicaId | BackendId | SchemaHash | Version | VersionHash | LstSuccessVersion | LstSuccessVersionHash | LstFailedVersion | LstFailedVersionHash | LstFailedTime | DataSize | RowCount | State  | LstConsistencyCheckTime | CheckVersion | CheckVersionHash | VersionCount | PathHash | MetaUrl                                           | CompactionStatus                                                |
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
| 245484   | 245485    | 10076     | 0          | 70      | 0           | 70                | 0                     | -1               | 0                    | <null>        | 2.5MB    | 54264    | NORMAL | <null>                  | -1           | 0                | 6            | 0        | http://172.26.197.103:8040/api/meta/header/245484 | http://172.26.197.103:8040/api/compaction/show?tablet_id=245484 |
+----------+-----------+-----------+------------+---------+-------------+-------------------+-----------------------+------------------+----------------------+---------------+----------+----------+--------+-------------------------+--------------+------------------+--------------+----------+---------------------------------------------------+-----------------------------------------------------------------+
```

Considering the actual number of tablets is 10,  after this PR, the version should be reduced from 5000 to 70

cost time:
before this PR: (17:09:26 ~ 18:05:05)，55min
after this PR: (19:05:05 ~ 19:12:53)，7min
```
MySQL (none)> show analyze status;
+--------+----------+----------------+---------+------+----------+---------+---------------------+---------------------+------------+------------------------------------------------------------------+
| Id     | Database | Table          | Columns | Type | Schedule | Status  | StartTime           | EndTime             | Properties | Reason                                                           |
+--------+----------+----------------+---------+------+----------+---------+---------------------+---------------------+------------+------------------------------------------------------------------+
| 191198 | hk_test  | test_0         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:09:26 | 2023-04-11 17:12:05 | {}         |                                                                  |
| 193783 | hk_test  | test_1         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:12:06 | 2023-04-11 17:14:45 | {}         |                                                                  |
| 196368 | hk_test  | test_2         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:14:45 | 2023-04-11 17:17:24 | {}         |                                                                  |
| 198953 | hk_test  | test_3         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:17:24 | 2023-04-11 17:20:02 | {}         |                                                                  |
| 201538 | hk_test  | test_4         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:20:03 | 2023-04-11 17:22:42 | {}         |                                                                  |
| 204123 | hk_test  | test_5         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:22:42 | 2023-04-11 17:25:20 | {}         |                                                                  |
| 206708 | hk_test  | test_6         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:25:20 | 2023-04-11 17:27:59 | {}         |                                                                  |
| 209293 | hk_test  | test_7         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:27:59 | 2023-04-11 17:30:38 | {}         |                                                                  |
| 211878 | hk_test  | test_8         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:30:38 | 2023-04-11 17:33:17 | {}         |                                                                  |
| 214463 | hk_test  | test_9         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:33:17 | 2023-04-11 17:35:56 | {}         |                                                                  |
| 217048 | hk_test  | test_10        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:35:56 | 2023-04-11 17:38:34 | {}         |                                                                  |
| 219633 | hk_test  | test_11        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:38:35 | 2023-04-11 17:41:13 | {}         |                                                                  |
| 222218 | hk_test  | test_12        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:41:14 | 2023-04-11 17:43:52 | {}         |                                                                  |
| 224803 | hk_test  | test_13        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:43:52 | 2023-04-11 17:46:31 | {}         |                                                                  |
| 227388 | hk_test  | test_14        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:46:31 | 2023-04-11 17:49:09 | {}         |                                                                  |
| 229973 | hk_test  | test_15        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:49:09 | 2023-04-11 17:51:50 | {}         |                                                                  |
| 232558 | hk_test  | test_16        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:51:51 | 2023-04-11 17:54:29 | {}         |                                                                  |
| 235143 | hk_test  | test_17        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:54:30 | 2023-04-11 17:57:08 | {}         |                                                                  |
| 237728 | hk_test  | test_18        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:57:09 | 2023-04-11 17:59:47 | {}         |                                                                  |
| 240313 | hk_test  | test_19        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 17:59:47 | 2023-04-11 18:02:26 | {}         |                                                                  |
| 242898 | hk_test  | test_20        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 18:02:27 | 2023-04-11 18:05:05 | {}         |                                                                  |
| 246002 | hk_test  | test_0         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:05:05 | 2023-04-11 19:05:31 | {}         |                                                                  |
| 246006 | hk_test  | test_1         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:05:31 | 2023-04-11 19:05:54 | {}         |                                                                  |
| 246010 | hk_test  | test_2         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:05:54 | 2023-04-11 19:06:16 | {}         |                                                                  |
| 246014 | hk_test  | test_3         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:06:16 | 2023-04-11 19:06:38 | {}         |                                                                  |
| 246018 | hk_test  | test_4         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:06:38 | 2023-04-11 19:07:00 | {}         |                                                                  |
| 246022 | hk_test  | test_5         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:07:00 | 2023-04-11 19:07:22 | {}         |                                                                  |
| 246026 | hk_test  | test_6         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:07:22 | 2023-04-11 19:07:44 | {}         |                                                                  |
| 246030 | hk_test  | test_7         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:07:44 | 2023-04-11 19:08:06 | {}         |                                                                  |
| 246034 | hk_test  | test_8         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:08:06 | 2023-04-11 19:08:28 | {}         |                                                                  |
| 246038 | hk_test  | test_9         | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:08:28 | 2023-04-11 19:08:50 | {}         |                                                                  |
| 246042 | hk_test  | test_10        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:08:50 | 2023-04-11 19:09:12 | {}         |                                                                  |
| 246046 | hk_test  | test_11        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:09:12 | 2023-04-11 19:09:34 | {}         |                                                                  |
| 246050 | hk_test  | test_12        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:09:34 | 2023-04-11 19:09:56 | {}         |                                                                  |
| 246054 | hk_test  | test_13        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:09:56 | 2023-04-11 19:10:18 | {}         |                                                                  |
| 246058 | hk_test  | test_14        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:10:18 | 2023-04-11 19:10:40 | {}         |                                                                  |
| 246062 | hk_test  | test_15        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:10:40 | 2023-04-11 19:11:02 | {}         |                                                                  |
| 246066 | hk_test  | test_16        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:11:03 | 2023-04-11 19:11:24 | {}         |                                                                  |
| 246070 | hk_test  | test_17        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:11:25 | 2023-04-11 19:11:47 | {}         |                                                                  |
| 246074 | hk_test  | test_18        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:11:47 | 2023-04-11 19:12:09 | {}         |                                                                  |
| 246078 | hk_test  | test_19        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:12:09 | 2023-04-11 19:12:31 | {}         |                                                                  |
| 246082 | hk_test  | test_20        | ALL     | FULL | ONCE     | SUCCESS | 2023-04-11 19:12:31 | 2023-04-11 19:12:53 | {}         |                                                                  |
+--------+----------+----------------+---------+------+----------+---------+---------------------+---------------------+------------+------------------------------------------------------------------+
55 rows in set
Time: 0.033s
````

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/runtime/statistic_result_writer.cpp [hunk 1]
```diff
@@ -281,6 +285,61 @@ Status StatisticResultWriter::_fill_table_statistic_data(int version, const Colu
...
 }
 
+/*
+FE SQL:    
+    SELECT cast(4 as INT), 
+         cast(7164015 as BIGINT), 
+         'k1', 
+         cast(COUNT(1) as BIGINT), 
+         cast(COUNT(1) * 4 as BIGINT), 
+         hll_serialize(IFNULL(hll_raw(`k1`), hll_empty())),
```
Comment (kangkaisen) [inline be/src/runtime/statistic_result_writer.cpp:295]: why hll_raw still need hll_serialize?

Comment (Seaven) [reply be/src/runtime/statistic_result_writer.cpp:295]: we need send the hll's data to fe, soo need serialize function

File: fe/fe-core/src/main/java/com/starrocks/statistic/FullStatisticsCollectJob.java [hunk 2]
```diff
@@ -66,12 +98,137 @@ public void collect(ConnectContext context, AnalyzeStatus analyzeStatus) throws
...
+        int maxRetryTimes = 5;
+        StatementBase insertStmt = createInsertStmt();
+        do {
+            // don't work if not leader, because origin statement is a mock SQL and can't forward to leader
+            if (!(GlobalStateMgr.getCurrentState().isLeader())) {
+                LOG.info("Statistics collect fail because this isn't leader");
+                return;
+            }
+
+            LOG.debug("statistics insert sql : " + insertStmt.getOrigStmt().originStmt);
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/statistic/FullStatisticsCollectJob.java:198]: Is necessary to print the mocked sql? It's too long. We'd better print some necessary info?

File: fe/fe-core/src/main/java/com/starrocks/statistic/FullStatisticsCollectJob.java [hunk 3]
```diff
@@ -66,12 +98,137 @@ public void collect(ConnectContext context, AnalyzeStatus analyzeStatus) throws
...
+        if (rowsBuffer.isEmpty()) {
+            return;
+        }
+
+        int count = 0;
+        int maxRetryTimes = 5;
+        StatementBase insertStmt = createInsertStmt();
+        do {
+            // don't work if not leader, because origin statement is a mock SQL and can't forward to leader
+            if (!(GlobalStateMgr.getCurrentState().isLeader())) {
```
[the remark this answers is not in the document]
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/statistic/FullStatisticsCollectJob.java:193]: yeah，it's only work on leader, I just worry too much

File: fe/fe-core/src/main/java/com/starrocks/statistic/FullStatisticsCollectJob.java (hunk 2 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/statistic/FullStatisticsCollectJob.java:198]: To be honest, I didn't think too clearly, it's just used for my verification work

File: be/src/runtime/statistic_result_writer.cpp (hunk 1 above)
Comment (kangkaisen) [reply be/src/runtime/statistic_result_writer.cpp:295]: why could we directly send the hll data to FE? which should work.

**Review by kangkaisen — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

File: be/src/runtime/statistic_result_writer.h [hunk 4]
```diff
@@ -54,6 +54,9 @@ class StatisticResultWriter final : public ResultWriter {
     Status _fill_table_statistic_data(int version, const Columns& columns, const Chunk* chunk,
                                       TFetchDataResult* result);
 
+    Status _fill_full_statistic_data_v4(int version, const Columns& columns, const Chunk* chunk,
```
[the remark this answers is not in the document]
Comment (Seaven) [reply be/src/runtime/statistic_result_writer.h:57]: match with STATISTICS_CONSTANT.version

**Pull request merged**